### PR TITLE
Upgrade location services

### LINKED ISSUE #510: Overhaul/update location services troubleshooting
Permissions as a whole are going to be a pain, since they depend on the browser having access, and then the website.

Audit what we have; see what options we have for fixing things (are there websites that guide the user on how to fix it?)

### PR DESCRIPTION
**Pull request by DeeTheDev**
Fixes #510

Refactor/overhaul the way we help our members when there is a location service error.
- Remove current manual (unlocalized) instructions modal
- Rename component to `LocationInstructionsAlert`
- Add supported browser instructions externalLinks (rather than localizing them into strings)
- Ensure we redirect with the correct language
- Update en/es location alert translations
- Update overlay width and padding to match the rest of the UI
- Display a warning if browser/device was not found
- Fix translation key typo

---

Created `SumaButton` with recommended stylings by our designer Anne. It is used in the Mobility components but needs to be applied globally where needed.

---

### Updated location alert for every browser/device
Screenshot 2023-09-28 at 7 02 31 PM

### Spanish location alert example
Screenshot 2023-09-28 at 7 03 27 PM

### Location instructions not found example
Screenshot 2023-09-28 at 7 06 32 PM

### Location instructions and zero-balance account warnings
Screenshot 2023-09-28 at 7 08 30 PM

### Click on vehicle error and zero-balance account warning
Screenshot 2023-09-22 at 9 36 58 PM

### Updated Mobility UI spacing
Screenshot 2023-09-25 at 11 47 27 AM

### Mobility reservation card global button styling example
Screenshot 2023-09-28 at 7 10 00 PM

### CODE REVIEW

File: webapp/public/locale/en/strings.json [hunk 1]
```diff
@@ -240,15 +238,19 @@
...
+    "activate_location_warning": "Activate Location",
+    "add_funds_warning": "Add money to your account before using suma's Mobility Services.",
     "do_not_park_intro": "You cannot park in this area. We may take action for repeated violations.",
     "do_not_park_title": "Please do not park here",
     "do_not_ride_intro": "You cannot ride in this area. We may take action for repeated violations.",
     "do_not_ride_title": "Please do not ride here",
     "end_trip": "End Trip",
     "intro": "Ride with enhanced privacy and better prices.",
     "locate_me": "Locate me",
+    "location_instructions_missing": "We could not find instructions to activate location services on your browser. Contact your administrator for more help.",
```
Comment (rgalanakis) [inline webapp/public/locale/en/strings.json:250]: Render in the help email or number here. If this is a thing we ever expect someone to see (and because we don't control their phones, someone will see it), it needs to be useful. There are some places, like private accounts, we can use language like this because no one will ever see it based on what they do/their phone (they'd only see it if we mess up).

File: webapp/src/components/mobilitymap/LocationInstructionsAlert.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    if (!isEmpty(userAgent)) {
+      return;
+    }
+    api
+      .getUserAgent()
+      .then(api.pickData)
+      .then((ua) => {
+        // get the second word in case for "Microsoft Edge"
+        const browser = ua.device.toLowerCase().split(" ")[1] || ua.device.toLowerCase();
+        if (ua.isIos) {
```
Comment (rgalanakis) [inline webapp/src/components/mobilitymap/LocationInstructionsAlert.js:28]: When do we see the browser-specific icons/instructions? This seems like we should be leading with the browser instructions, we can't know if the issue is the browser or the website.

File: webapp/public/locale/en/strings.json (hunk 1 above)
Comment (DeeTheDev) [reply webapp/public/locale/en/strings.json:250]: I was thinking that phone number would be better but I don't know that number yet. I added the email.

File: webapp/src/components/mobilitymap/LocationInstructionsAlert.js (hunk 2 above)
Comment (DeeTheDev) [reply webapp/src/components/mobilitymap/LocationInstructionsAlert.js:28]: @rgalanakis I see what you mean. How about we always provide browser link if it's available, but also provide the device/platform (Android, IOS macOs, windows) link if we find one?

Comment (rgalanakis) [reply webapp/src/components/mobilitymap/LocationInstructionsAlert.js:27]: I think we need to show both :( Unless the links cover both.

Comment (DeeTheDev) [reply webapp/src/components/mobilitymap/LocationInstructionsAlert.js]: @rgalanakis Good catch. The condition totally changed and made me realize that multiple externalLinks were created unnecessarily.
I added a commit for this.

File: webapp/public/locale/en/strings.json [hunk 3]
```diff
@@ -1,6 +1,5 @@
 {
   "common": {
-    "add_money_to_account": "Press here to add money to your account",
```
Comment (rgalanakis) [inline webapp/public/locale/en/strings.json:3]: Why was this removed?

File: webapp/src/components/mobilitymap/Map.js [hunk 4]
```diff
@@ -162,7 +164,13 @@ const Map = () => {
         <CardOverlay>
           <FormError error={error} noMargin component="div" />
           {readOnlyReason(user, "read_only_zero_balance") && (
-            <Link to="/funding">{t("common:add_money_to_account")}</Link>
+            <Alert className={clsx("mb-0", Boolean(error) && "mt-3")} variant="warning">
+              <p>
+                <i className="bi bi-exclamation-triangle-fill me-1"></i>
+                {t("mobility:add_funds_warning")}
+              </p>
+              <SumaButton href="/funding">{t("forms:add_funds")}</SumaButton>
```
Comment (rgalanakis) [inline webapp/src/components/mobilitymap/Map.js:172]: What is a 'suma button'? Why can't we use one of the existing button classes? I'm confused what is going on here.

File: webapp/src/components/mobilitymap/LocationInstructionsAlert.js [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+ * Bootstrap icon className.
+ *
+ * Some browsers and platforms will have the same links but it is okay.
+ *
+ * Only the found browser and/or platform returned.
+ *
+ * @param ua The user agent to analyze device, platform and other data
+ * @returns []
+ */
+function supportedBrowserAndPlatform(ua) {
```
Comment (rgalanakis) [inline webapp/src/components/mobilitymap/LocationInstructionsAlert.js:71]: @DeeTheDev we have an endpoint that already parses and returns structured information about the browser. Why would we want to do further string manipulation? Modify the backend to return the structure data we need rather than all of this mangling and manipulation.

File: webapp/src/modules/externalLinks.js [hunk 6]
```diff
@@ -1,8 +1,21 @@
 const externalLinks = {
   safeHosts: ["https://mysuma.org"],
   mobilityInfoLink: "https://mysuma.org/sumaplatform#mobility",
-  sumaIntroLink: "https://mysuma.org/",
   privacyPolicy: "https://app.mysuma.org/privacy-policy",
+  sumaIntroLink: "https://mysuma.org/",
+  edge: (lang) =>
+    `https://support.microsoft.com/${lang}-us/windows/windows-location-service-and-privacy-3a8eee0a-5b0b-dc07-eede-2a5ca1c49088`,
```
Comment (rgalanakis) [inline webapp/src/modules/externalLinks.js:7]: Is `es-us` a language?

It's not- this link happens to work because MS redirects it. In fact the entire URL changes. We should be depending on broken links.

File: webapp/src/assets/styles/mobility.scss [hunk 7]
```diff
@@ -1,5 +1,7 @@
 @import "./theme";
 
+$overlay-padding-x-size: 48px;
```
Comment (rgalanakis) [inline webapp/src/assets/styles/mobility.scss:3]: Why is this a variable? Why is this hard-coded to 48px and not based on spacing?

**Review by rgalanakis — CHANGES_REQUESTED** [review]
Dee, this needs quite a bit of work still. There are some failures of judgement. It can be useful to look back at work like this the day after you put up a PR, and consider whether the choices make sense in the light of a new day.

Please hold off on any modifications to this until we pair on it.

File: webapp/public/locale/en/strings.json (hunk 3 above)
Comment (DeeTheDev) [reply webapp/public/locale/en/strings.json:3]: It was replaced with an alert component and context with an "Add funds" button.

File: webapp/src/components/mobilitymap/Map.js (hunk 4 above)
Comment (DeeTheDev) [reply webapp/src/components/mobilitymap/Map.js:172]: Not sure about the name either but Anne has commented about the use of different button styles on some of the app pages and advises to set a general button across the app. I'm trying to start that here. We could omit it for now
.

File: webapp/src/components/mobilitymap/LocationInstructionsAlert.js (hunk 5 above)
Comment (DeeTheDev) [reply webapp/src/components/mobilitymap/LocationInstructionsAlert.js:71]: Thanks for this. Will remember to double check if we have available functionality

**Comment by rgalanakis** [comment]
Superceded by #516, may need this in the future when we have MaaS-based mobility functionality.
